boards/BMS/Inc/App/App_Soc.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "App_SharedExitCode.h"
+
+/**
+ * Given three state-of-charge (SoCs), check whether the difference between any
+ * two SoCs is less than or equal to the specified threshold. If any two SoCs
+ * pass the check, return the average value of those two SoCs.
+ * @param max_difference The maximum tolerable difference between any two given
+ *                       state-of-charge values
+ * @param soc_1 The first SoC
+ * @param soc_2 The second SoC
+ * @param soc_3 The third SoC
+ * @param result The average value of two SoCs whose difference is less than or
+ *               equal to max_value, else NaN. In the event of a tiebreak (i.e.
+ *               All three SoCs pass the check), th first two SoCs are used to
+ *               calculate the average value.
+ * @return EXIT_CODE_INVALID_ARGS if the max difference or any of the SoCs is
+ *                                less than 0 or greater than 100
```

Comment:
should `100` here be replaced with `max_difference`? Also I'm not sure that returning `NaN` to indicate no value is the best option. Perhaps we could just set the `result` pointer to `NULL`?

Reply:
It should be `100`. I rewrote most of the comment so hopefully it's all more clear now. As for the `NULL` suggestion, the user is providing the storage so setting it to NULL would be the same as returning 0.0f I think?

Reply:
ah yeah, it would be. My bad, this looks fine.